crates/holochain/src/conductor/cell.rs

```diff
@@ -805,6 +807,9 @@ impl Cell {
     /// Check if each Zome's init callback has been run, and if not, run it.
     #[tracing::instrument(skip(self))]
     async fn check_or_run_zome_init(&self) -> CellResult<()> {
+        // Ensure that only one init check is run at a time
+        let _guard = self.init_mutex.lock().await;
```

Comment:
Do you think this is worth explicitly timing out, so we don't get forever deadlocks?

```suggestion
        let _guard = tokio::time::timeout(std::time::Duration::from_secs(30), self.init_mutex.lock()).await?;
```

Reply:
@neonphog I'm on the fence. Yes, if an init function locks up, then it will lock up the Cell. But, if the init function can't complete, the Cell is unusable anyway. And what if somebody really wants to do several minutes of PoW during init?

Yeah, I guess I'm leaning more towards having a timeout just to raise visibility into a potentially locked-up init function. And it's reasonable to require init to complete in a certain amount of time. I'm not sure what it should be though. Is 10 minutes too long? I don't know if our current wasmer setup includes bad consequences for having a Wasm instance running for too long.

Reply:
OK I realized it's not actually a timeout on init, but on waiting for another function to finish init, which makes even more sense. I added a 30s timeout like you suggested.